js/src/logger.ts

```diff
@@ -3474,7 +3572,7 @@ function deepCopyEvent<T extends Partial<BackgroundLogEvent>>(event: T): T {
       return `<dataset>`;
     } else if (v instanceof Logger) {
       return `<logger>`;
-    } else if (v instanceof Attachment) {
+    } else if (v instanceof Attachment || v instanceof ExternalAttachment) {
```

Comment:
I think you could have a if you move the interface to an abstract class you could avoid the `|| instanceof`